ktor-client/ktor-client-core/posix/src/io/ktor/client/utils/CoroutineUtilsPosix.kt

```diff
@@ -15,3 +16,16 @@ public actual fun Dispatchers.clientDispatcher(
     threadCount: Int,
     dispatcherName: String
 ): CoroutineDispatcher = Unconfined
+
+internal actual fun checkCoroutinesVersion() {
+    try {
+        val parent = Job()
+        parent.freeze()
+        Job(parent)
+    } catch (cause: Throwable) {
+        val message = "Invalid coroutines version. " +
+            "Consider using version with suffix `-native-mt`, like `1.3.9-native-mt` for native."
+
+        error(message)
```

Comment:
Shouldn't we use some specific exception, error or even crash?

Reply:
I'm not sure the purpose of the `specific` exception here: what is the reason for catching it?